autolab/Autolab
# Assessment direct handout download

## Summary

### Summary generated by Reviewpad on 02 Jan 24 19:03 UTC
This pull request adds a new feature to the code. It introduces a "handout_only" toggle to the assessment model and views. The toggle allows an assessment to be marked as consisting solely of a handout, which will be directly downloaded when a student clicks on the assessment. The patch also includes updates to the schema.rb file and wording changes in the basic assessment edit view. Additionally, there are some undoing of inadvertent regressions in the index view. Overall, this pull request enhances the functionality of the assessment module.


## Description

Add a "handout only" option to assessments. When enabled, non-instructor users who click on the assessment will be directed to the handout.

## Motivation and Context


Facilitates the use of Autolab as a resource sharing medium (requested by 122).

Closes #2021 

## How Has This Been Tested?




Enable handout only option (under basic) and set a handout.

![Screenshot 2023-12-03 at 22 19 55](https://github.com/autolab/Autolab/assets/9074856/1f74e1da-3cc8-4fca-9022-1675c187fffd)
<img width="989" alt="Screenshot 2024-01-05 at 02 19 56" src="https://github.com/autolab/Autolab/assets/9074856/77c47e51-c9e5-4c81-90e9-6c8988d1cacf">

- As an instructor, clicking on the assessment loads it as usual.
- As a student, clicking on the assessment brings you directly to the handout.
- Ensure `handout_only` setting is dumped to assessment yml file
- Ensure `handout_only` setting is imported. e.g. delete the assessment and import from file system. Option should still be set.

## Types of changes


## Checklist:
